### PR TITLE
AAKD, on kili-playground, when I look at the API, I want that jsonResponseContains is not used

### PR DESCRIPTION
**Pull request by theodu**
- [ ] I opened a merge request on `kili` on a branch with the same name than the branch of the current pull request and forced security tests.
- [ ] I opened a pull request on the Node playground if the changes are breaking.

### CODE REVIEW

File: kili/queries/label/__init__.py [hunk 1]
```diff
@@ -95,6 +98,18 @@ def labels(self,
         Examples:
             >>> kili.labels(project_id=project_id, fields=['jsonResponse', 'labelOf.externalId']) # returns a list of all labels of a project and their assets external ID
             >>> kili.labels(project_id=project_id, fields=['jsonResponse'], as_generator=True) # returns a generator of all labels of a project
+
+        !!! example "How to filter based on label categories"
+            The search query is composed of logic expressions following this format: `[Job_name].[Category_name].count [Comparaison operator] [Value]` where:
```
Comment (mDuval1) [inline kili/queries/label/__init__.py:103]: typo (Comparison)

Comment (mDuval1) [reply kili/queries/label/__init__.py:103]: logic -> logical (adjective)

File: kili/queries/label/__init__.py [hunk 2]
```diff
@@ -95,6 +98,18 @@ def labels(self,
...
+            - Job_name is the name of the job in the interface
+            - Category_name is the name of the category in the interface for this job
+            - Comparaison operators can be : [`==`, `>=`, `<=`, `<`, `>`]
+            - Value is an interger that represent the count of such object of the given category in the label
```
[the remark this answers is not in the document]
Comment (mDuval1) [reply kili/queries/label/__init__.py:107]: such objects

File: kili/helpers.py [hunk 3]
```diff
@@ -332,3 +333,25 @@ def infer_id_from_external_id(kili, asset_id: str, external_id: str, project_id:
...
+    condition = identifier + operator + number
+
+    expr = pp.infixNotation(condition, [
+        ("AND", 2, pp.opAssoc.LEFT, ),
+        ("OR", 2, pp.opAssoc.LEFT, ),
+    ])
+    try:
+        expr.parseString(query, parseAll=True)
+    except pp.ParseException as error:
+        raise ValueError(f'Invalid category search query: {query}') from error
```
Comment (mDuval1) [inline kili/helpers.py:357]: you should return the error message of pp

Comment (theodu) [reply kili/helpers.py:357]: This is what the `from error`is for. 
Example with: `kili.assets(project_id=project_id, label_category_search = "JOB_0.CLASS_B")`

```---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
~/anaconda3/envs/python3/lib/python3.9/site-packages/pyparsing.py in _parseNoCache(self, instring, loc, doActions, callPreParse)
   1682                 try:
-> 1683                     loc, tokens = self.parseImpl(instring, preloc, doActions)
   1684                 except IndexError:

~/anaconda3/envs/python3/lib/python3.9/site-packages/pyparsing.py in parseImpl(self, instring, loc, doActions)
   2896     def parseImpl(self, instring, loc, doActions=True):
-> 2897         if instring[loc] == self.firstMatchChar:
   2898             return loc + 1, self.match

IndexError: string index out of range

During handling of the above exception, another exception occurred:

ParseException                            Traceback (most recent call last)
~/kili-python-sdk/kili/helpers.py in parse_category_search_query(query)
    354     try:
--> 355         expr.parseString(query, parseAll=True)
    356     except pp.ParseException as error:

~/anaconda3/envs/python3/lib/python3.9/site-packages/pyparsing.py in parseString(self, instring, parseAll)
   1954                     exc.__traceback__ = self._trim_traceback(exc.__traceback__)
-> 1955                 raise exc
   1956         else:

~/anaconda3/envs/python3/lib/python3.9/site-packages/pyparsing.py in _parseNoCache(self, instring, loc, doActions, callPreParse)
   1684                 except IndexError:
-> 1685                     raise ParseException(instring, len(instring), self.errmsg, self)
   1686             else:

ParseException: Expected {OR term | AND term}, found end of text  (at char 13), (line:1, col:14)

The above exception was the direct cause of the following exception:

ValueError                                Traceback (most recent call last)
<ipython-input-3-0c57c8dd9e9c> in <module>
----> 1 kili.assets(project_id='cl3d45bzb0rf70mx4580m0bt1', label_category_search = "JOB_0.CLASS_B")

~/kili-python-sdk/kili/helpers.py in checked_resolver(*args, **kwargs)
     53                     f' {resolver.__name__} with arguments {args}') from exception
     54             if self.client_is_compatible(client_endpoint):
---> 55                 return resolver(*args, **kwargs)
     56             raise EndpointCompatibilityError(
     57                 resolver.__name__, client_endpoint)

~/anaconda3/envs/python3/lib/python3.9/site-packages/typeguard/__init__.py in wrapper(*args, **kwargs)
    808         memo = _CallMemo(python_func, _localns, args=args, kwargs=kwargs)
    809         check_argument_types(memo)
--> 810         retval = func(*args, **kwargs)
    811         check_return_type(retval, memo)
    812 

~/kili-python-sdk/kili/helpers.py in wrapper(*args, **kwargs)
    251             if msg:
    252                 warnings.warn(msg, _type, stacklevel=2)
--> 253             return func(*args, **kwargs)
    254         return wrapper
    255     return decorator

~/kili-python-sdk/kili/queries/asset/__init__.py in assets(self, asset_id, project_id, skip, fields, asset_id_in, consensus_mark_gt, consensus_mark_lt, disable_tqdm, external_id_contains, first, format, honeypot_mark_gt, honeypot_mark_lt, label_author_in, label_consensus_mark_gt, label_consensus_mark_lt, label_created_at, label_created_at_gt, label_created_at_lt, label_honeypot_mark_gt, label_honeypot_mark_lt, label_json_response_contains, label_type_in, metadata_where, skipped, status_in, updated_at_gte, updated_at_lte, as_generator, label_category_search)
    164         disable_tqdm = disable_tqdm or as_generator
    165         if label_category_search:
--> 166             parse_category_search_query(label_category_search)
    167 
    168         payload_query = {

~/kili-python-sdk/kili/helpers.py in parse_category_search_query(query)
    355         expr.parseString(query, parseAll=True)
    356     except pp.ParseException as error:
--> 357         raise ValueError(f'Invalid category search query: {query}') from error

ValueError: Invalid category search query: JOB_0.CLASS_B```

Comment (theodu) [reply kili/helpers.py:357]: @mDuval1

**Review by p-desaintchamas — APPROVED** [review]

File: kili/helpers.py [hunk 4]
```diff
@@ -332,3 +333,25 @@ def infer_id_from_external_id(kili, asset_id: str, external_id: str, project_id:
...
+
+
+def parse_category_search_query(query):
+    """Parse the category search query to assess its validity
+    Args:
+        query: the query to parse
+    """
+    operator = pp.oneOf(">= <= > < ==")
+    number = pp.pyparsing_common.number()
+    dot = '.'
```
[the remark this answers is not in the document]
Comment (PierreLeveau) [reply kili/helpers.py:345]: (3 lines above)

File: kili/helpers.py [hunk 5]
```diff
@@ -332,3 +333,25 @@ def infer_id_from_external_id(kili, asset_id: str, external_id: str, project_id:
             f'Several assets found containing external ID "{external_id}":'
             f' {assets}. Please, use asset ID instead.')
     return assets[0]['id']
+
+
+def parse_category_search_query(query):
```
Comment (PierreLeveau) [inline kili/helpers.py:338]: -> `validate_category_search_query`

File: kili/helpers.py [hunk 6]
```diff
@@ -332,3 +333,25 @@ def infer_id_from_external_id(kili, asset_id: str, external_id: str, project_id:
...
+    """Parse the category search query to assess its validity
```
Comment (PierreLeveau) [inline kili/helpers.py:339]: `Validates the category search query (throws an exception if invalid)`

File: kili/queries/asset/__init__.py [hunk 7]
```diff
@@ -132,6 +139,24 @@ def assets(self,
...
+            - `[job_name]` is the name of the job in the interface
+            - `[category_name]` is the name of the category in the interface for this job
+            - `[comparaison_operator]` can be one of: [`==`, `>=`, `<=`, `<`, `>`]
+            - `[value]` is an integer that represents the count of such objects of the given category in the label
+
+            These operations can be separated by OR and AND operators
+
+            Example:
+
+                label_category_search = `JOB_0.OBJECT_A.count > 0`
```
Comment (PierreLeveau) [inline kili/queries/asset/__init__.py:157]: `OBJECT_A` => `CATEGORY_A`, object is confusing I think

File: kili/queries/asset/__init__.py [hunk 8]
```diff
@@ -282,9 +312,9 @@ def count_assets(self, asset_id: Optional[str] = None,
                 or an error message.
 
         Examples:
-            >>> kili.count_assets(project_id=project_id)
+            >> > kili.count_assets(project_id=project_id)
```
Comment (PierreLeveau) [inline kili/queries/asset/__init__.py:315]: typo ?

File: kili/queries/asset/__init__.py [hunk 9]
```diff
@@ -132,6 +139,24 @@ def assets(self,
...
+
+            - `[job_name]` is the name of the job in the interface
+            - `[category_name]` is the name of the category in the interface for this job
+            - `[comparaison_operator]` can be one of: [`==`, `>=`, `<=`, `<`, `>`]
+            - `[value]` is an integer that represents the count of such objects of the given category in the label
+
+            These operations can be separated by OR and AND operators
+
+            Example:
+
```
Comment (PierreLeveau) [inline kili/queries/asset/__init__.py:156]: Add a comment for each line (especially to show that the classification case is handled)

**Review by PierreLeveau — CHANGES_REQUESTED** [review]

File: kili/helpers.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (theodu) [reply kili/helpers.py:345]: which line ? the operator one?

**Comment by PierreLeveau** [comment]
I tested the `label_category_search` on several projects. The query was accepted but returned no result. Happy to provide the debug info.

**Comment by PierreLeveau** [comment]
My bad, I did the query with classes not present in the dataset.

**Review by PierreLeveau — APPROVED** [review]

**Pull request merged**